bakery/checkers/declared.go

```diff
@@ -76,33 +77,42 @@ func (c Declared) Check(_, arg string) error {
 // different values, the information is omitted from the map.
 // When the caveats are later checked, this will cause the
 // check to fail.
-func InferDeclared(ms macaroon.Slice) Declared {
-	var conflicts []string
-	info := make(Declared)
+func InferDeclared(ns *Namespace, ms macaroon.Slice) map[string]string {
```

Comment:
Conditions with Location=="" are the first party caveats. I don't see what the issue is.